core/primitives/src/utils.rs

```diff
@@ -35,89 +35,250 @@ pub mod col {
     pub const DELAYED_RECEIPT: &[u8] = &[8];
 }
 
-fn key_for_column_account_id(column: &[u8], account_key: &AccountId) -> Vec<u8> {
-    let mut key = Vec::with_capacity(column.len() + account_key.len());
-    key.extend(column);
-    key.extend(account_key.as_bytes());
-    key
-}
+#[derive(derive_more::AsRef, derive_more::Into)]
+struct KeyForColumnAccountId(Vec<u8>);
```

Comment:
I had to refactor this file as I was adding a few more functions and already felt like loosing track of them, so I essentially grouped the functions into namespaces. /cc @mikhailOK it seems that you have been working on these before. Please, review the changes.